localstack/packages/api.py

```diff
@@ -72,15 +76,18 @@ def install(self, target: Optional[InstallTarget] = None) -> None:
         try:
             if not target:
                 target = InstallTarget.VAR_LIBS
-            with self.lock:
-                if not self.is_installed():
-                    LOG.debug("Starting installation of %s...", self.name)
-                    self._prepare_installation(target)
-                    self._install(target)
-                    self._post_process(target)
-                    LOG.debug("Installation of %s finished.", self.name)
-                else:
-                    LOG.debug("Installation of %s skipped (already installed).", self.name)
+            # Skip the installation if it's already installed
+            if not self.is_installed():
+                with self.install_lock:
+                    # Package might have been installed in the meantime, check again
+                    if not self.is_installed():
```

Comment:
I think a single `self.is_installed()` check inside the lock context should suffice

Reply:
But this would mean that a check if a package is installed can be blocked by another installer.
For example, two installers can share a lock because they use globally exclusive tooling when installing (like os level package managers).
This would mean that one installer can not check if it's already installed for the whole execution time of the second installer (because it holds the shared lock).

Reply:
I see, two installers for different versions could share the same lock and that's when the is installed check will be blocked, when not actually necessary. :+1: